hypothesis-python/src/hypothesis/database.py

```diff
@@ -146,10 +146,47 @@ def __call__(self, *args: Any, **kwargs: Any) -> "ExampleDatabase":
 
 
 class ExampleDatabase(metaclass=_EDMeta):
-    """An abstract base class for storing examples in Hypothesis' internal format.
+    """
+    A Hypothesis database, for use in the |settings.database| setting.
+
+    A Hypothesis database acts like a ``Mapping[bytes, set[bytes]]``, mapping
+    each ``bytes`` key to a set of ``bytes`` values.
```

Comment:
I think we want to open with a short motivation, and the type is really just as a "you can implement this" hint.  Something like

> Using an ExampleDatabase (via the database= setting) allows Hypothesis to remember and replay failing examples, and ensure that your tests are never flaky.  We provide several concrete subclasses, and you can write a custom dabase backed by your preferred way to store a `dict[bytes, set[bytes]]`.